src/main/haskell/kore/src/Kore/Predicate/Predicate.hs

```diff
@@ -279,3 +283,13 @@ makeFalsePredicate
     => Predicate level var
 makeFalsePredicate =
     GenericPredicate mkBottom
+
+{- | Replace all variables in a @Predicate@ using the provided mapping.
+-}
+mapVariables :: (from level -> to level) -> Predicate level from -> Predicate level to
+mapVariables f = wrapPredicate . mapPatternVariables f . unwrapPredicate
```

Comment:
Is `fmap (mapPatternVariables variableMapper) predicate` better?

Regardless of whether it's better, could you replace former uses of the fmap stuff  with your function? There should be at leas one of these in ExpandedPattern